docs/C API.md

```diff
@@ -0,0 +1,127 @@
+  to tables have infix `table_` and names of them will start with `nikss_table_`.
+- The last part of the names (postfix) denotes the purpose of the object (structures, functions, etc.), e.g. functions ended
+  with `free` will release all resources allocated for instance of the object.
+- All the functions operates on instances of structures or typedefs, called `contex` or `ctx` for short. Each context must
+  be initialized (`init` postfix) before use and freed when no more needed (`free` postfix).
+- The content of structures are not considered as a public API, can change at any time. Always use provided functions to
+  operate on instances of objects.
+- Instances of objects created by the API (returned as a pointer) also must be freed after used.
+- Functions in most cases return `0` (`NO_ERROR` constant) or a valid pointer (not `NULL`) on success. In case of error,
+  error code (values from POSIX standard are used) or `NULL` pointer is returned and some message on `stderr` might be printed
+  (some functions print to `stdout` but it is considered to change). Functions that return special values, like data length,
+  member reference, etc. have own error values.
+- Passing `NULL` pointer to any functions is safe (but not useful). Arguments are validated and error is returned if they
+  are required.
+- All instances of objects are movable, but not copyable. After move pointers acquired using given instance are invalid
+  (some such functionalities use memory space inside context).
+- Data passed to or from functions are considered to be plain binary in the host order byte.
+
+# Basic usage
+
+## Header files
+
+Table below shows which header have to be included to use API for required functionality.
+
+| Functionality                                                                                                                        | Header file         |
+|--------------------------------------------------------------------------------------------------------------------------------------|---------------------|
+| `ActionProfile`,<br/>`ActionSelector`,<br/>`Counter`,<br/>`DirectCounter`,<br/>`DirectMeter`,<br/>`Meter`,<br/>`Register`,<br/>Table | `nikss.h`           |
+| Clone session,<br/>Multicast group                                                                                                   | `nikss_pre.h`       |
+| `Digest`                                                                                                                             | `nikss_digest.h`    |
+| Pipeline and port management                                                                                                         | `nikss_pipeline.h`  |
+| `value_set`                                                                                                                          | `nikss_value_set.h` |
+
+## Pipeline
+
+To be able to manage any P4 object, pipeline context `nikss_context_t` is mandatory. Following use is minimal and has no
```

Comment:
```suggestion
The pipeline context `nikss_context_t` is mandatory to manage any P4 object. The following use is minimal and has no
```

Reply:
@tatry what does it mean that the use is "minimal"?

Reply:
I mean that `context` is ready for operation on P4 object but do nothing to it (like get entry or modify it).